docs/docs/examples/repl.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: docs
+title: REPL
+section: examples
+sort_order: 1001
+---
+
+REPL
+----
+
+This examples shows how to run OPA as an interactive shell or [REPL (read-eval-print loop)](https://en.wikipedia.org/wiki/Read–eval–print_loop).
+
+REPLs are great for learning new languages and running quick experiments. You can take advantage of OPA's REPL to quickly
+run ad-hoc queries or prototype policies.
+
+In this example, we will use the REPL to define rules to identify servers that violate a security policy: servers exposing (unencrypted) HTTP must not be connected to a public network.
+
+Once you finish this example, you will be familiar with:
+
+- Running OPA as an interactive shell/REPL.
+- Writing ad-hoc queries in [Rego](/docs/lang.html).
+
+Prerequisites
+=============
+
+- This example requires that you have the latest version of OPA. You can download the latest version of OPA at [openpolicyagent.org](/index.html)
+
+- This example assumes you have the OPA executable in your $PATH.
+
+Steps
+=====
+
+1. First, make sure you can run the REPL on your machine:
+
+        opa run
+
+    Without any data, you can experiment with simple boolean expressions to get the hang of it:
+
+        > a = 1, b = 2, a != b
+        +---+---+
+        | A | B |
+        +---+---+
+        | 1 | 2 |
+        +---+---+
+        > a = 1, b = 2, a = b
+        false
+        > 10 > 9
+        true
+        > a = [1,2,3,4], a[i] > 2
+        +-----------+---+
+        |     A     | I |
+        +-----------+---+
+        | [1,2,3,4] | 2 |
+        | [1,2,3,4] | 3 |
+        +-----------+---+
+
+    When you enter expressions into the REPL, you are effectively running *queries* against OPA. The REPL output shows the values of variables in the expression that make the query **true**. If there is not set of variables that would make the query true, the REPL prints **false**. If there are no variables in the query and the query evaluates successfully, then the REPL just prints **true**.
+
+1. In addition to running queries, the REPL also lets you define rules:
+
+        > p[x] :- a = [1,2,3,4], a[x] = _
+        defined
+        > p[x]
+        +---+
+        | X |
+        +---+
+        | 0 |
+        | 1 |
+        | 2 |
+        | 3 |
+        +---+
+
+1. The REPL can also take policies and data as input. Let's define a bit of JSON data that will be used in the example:
```

Comment:
Might be worth saying that you need to quit the REPL, run these commands from the command line, and restart OPA